pkg/auth/db_service.go

```diff
@@ -837,3 +868,372 @@ func (s *DBAuthService) markTokenSingleUse(ctx context.Context, tokenID string,
 	}
 	return canUseToken, nil
 }
+
+var ErrExportedEntNotFound = errors.New("previously exported entitiy not found")
+
+func Migrate(ctx context.Context, d *pgxpool.Pool, writer io.Writer) error {
+	je := json.NewEncoder(writer)
+	// Create header
+	if err := je.Encode(kv.Header{
+		LakeFSVersion:   version.Version,
+		PackageName:     model.PackageName,
+		DBSchemaVersion: kv.InitialMigrateVersion,
+		CreatedAt:       time.Now().UTC(),
+	}); err != nil {
+		return err
+	}
+
+	userIDToDetails, err := exportUsers(ctx, d, je)
+	if err != nil {
+		return err
+	}
+
+	groupIDToName, err := exportGroups(ctx, d, je)
+	if err != nil {
+		return err
+	}
+
+	policyIDToName, err := exportPolicies(ctx, d, je)
+	if err != nil {
+		return err
+	}
+
+	if err = exportUserGroups(ctx, d, je, userIDToDetails, groupIDToName); err != nil {
+		return err
+	}
+
+	if err = exportUserPolicies(ctx, d, je, userIDToDetails, policyIDToName); err != nil {
+		return err
+	}
+
+	if err = exportGroupPolicies(ctx, d, je, groupIDToName, policyIDToName); err != nil {
+		return err
+	}
+
+	if err = exportCredentials(ctx, d, je, userIDToDetails); err != nil {
+		return err
+	}
+
+	if err = exportExpiredTokens(ctx, d, je); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+type userDetails struct {
+	name string
+	kvID string
+}
+
+type UserIDToDetails map[int64]userDetails
+
+type IDToName map[int]string
+
+func exportUsers(ctx context.Context, d *pgxpool.Pool, je *json.Encoder) (UserIDToDetails, error) {
+	rows, err := d.Query(ctx, "SELECT * FROM auth_users ORDER BY created_at ASC")
+	if err != nil {
+		return nil, err
+	}
+	defer rows.Close()
+	userIDToDetails := make(UserIDToDetails)
+	scanner := pgxscan.NewRowScanner(rows)
+	for rows.Next() {
+		dbUser := model.DBUser{}
+		err = scanner.Scan(&dbUser)
+		if err != nil {
+			return nil, err
+		}
+		kvUser := &model.User{
+			ID:       uuid.New().String(),
+			BaseUser: dbUser.BaseUser,
+		}
+		key := model.UserPath(dbUser.Username)
+		value, err := proto.Marshal(model.ProtoFromUser(kvUser))
+		if err != nil {
+			return nil, err
+		}
+
+		if err = je.Encode(kv.Entry{
+			PartitionKey: []byte(model.PartitionKey),
+			Key:          []byte(key),
+			Value:        value,
+		}); err != nil {
+			return nil, err
+		}
+
+		userIDToDetails[dbUser.ID] = userDetails{
+			kvID: kvUser.ID,
```

Comment:
In the migration process - one of the things we need to do is to create a new UUID using a common source

Reply:
Seeanswer for `xid` below - let's handle it ina different issue